### PR TITLE
service: add timer support for systemd

### PR DESCRIPTION
**Pull request by SMillerDev**
- [x] Have you followed the guidelines in our [Contributing](https://github.com/Homebrew/brew/blob/HEAD/CONTRIBUTING.md) document?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/brew/pulls) for the same change?
- [x] Have you added an explanation of what your changes do and why you'd like us to include them?
- [x] Have you written new tests for your changes? [Here's an example](https://github.com/Homebrew/brew/blob/HEAD/Library/Homebrew/test/PATH_spec.rb).
- [ ] Have you successfully run `brew style` with your changes locally?
- [x] Have you successfully run `brew typecheck` with your changes locally?
- [x] Have you successfully run `brew tests` with your changes locally?

-----

Allows brew to generate systemd unit timers

### CODE REVIEW

File: Library/Homebrew/service.rb [hunk 1]
```diff
@@ -267,5 +277,28 @@ def to_systemd_unit
...
+      timer = <<~EOS
+        [Unit]
+        Description=Homebrew generated timer for #{@formula.name}
+
+        [Install]
+        WantedBy=timers.target
+
+        [Timer]
+        Persistent=true
+        Unit=#{@formula.service_name}
```
Comment (dawidd6) [inline Library/Homebrew/service.rb:294]:
```suggestion
```

https://www.freedesktop.org/software/systemd/man/systemd.timer.html#Unit=

Documentation says this parameter does not need to be set as it defaults to the timer name.

File: Library/Homebrew/service.rb [hunk 2]
```diff
@@ -267,5 +277,28 @@ def to_systemd_unit
...
+    def to_systemd_timer
+      timer = <<~EOS
+        [Unit]
+        Description=Homebrew generated timer for #{@formula.name}
+
+        [Install]
+        WantedBy=timers.target
+
+        [Timer]
+        Persistent=true
```
Comment (dawidd6) [inline Library/Homebrew/service.rb:293]: https://www.freedesktop.org/software/systemd/man/systemd.timer.html#Persistent=

> Note that this setting only has an effect on timers configured with OnCalendar=.

Should we use `OnCalendar` here instead? I guess that would be useful for us when the service for is cron.

File: Library/Homebrew/service.rb (hunk 1 above)
Comment (SMillerDev) [reply Library/Homebrew/service.rb:294]: Yeah I figured better safe than sorry

File: Library/Homebrew/service.rb (hunk 2 above)
Comment (SMillerDev) [reply Library/Homebrew/service.rb:293]: That's going to be my next PR. I'll make sure that this only applies on cron then

**Review by MikeMcQuaid — COMMENTED** [review]
Looks good so far. Do we expect similar support for `launchd`?

**Comment by SMillerDev** [comment]
> Looks good so far. Do we expect similar support for launchd?

We did, until https://github.com/Homebrew/brew/pull/12367

**Comment by MikeMcQuaid** [comment]
@SMillerDev Great, thanks!

**Comment by SMillerDev** [comment]
Can I merge this one?

**Review by MikeMcQuaid — APPROVED** [review]

**Comment by SMillerDev** [comment]
Rebasing and pushing in the hopes the review CI gets unstuck

**Pull request merged**